sktime/forecasting/compose/_pipeline.py

```diff
@@ -13,55 +13,17 @@
 from sktime.forecasting.base._sktime import _OptionalForecastingHorizonMixin
 from sktime.forecasting.base._sktime import _SktimeForecaster
 from sktime.transformations.base import _SeriesToSeriesTransformer
-from sktime.utils.validation.forecasting import check_y
+from sktime.utils.validation.forecasting import check_y_X, check_y, check_X
 from sktime.utils.validation.series import check_series
 from sktime.utils import _has_tag
 
 
-class TransformedTargetForecaster(
+class _Pipeline(
     _OptionalForecastingHorizonMixin,
```

Comment:
We need to adjust this to the new forecasting API introduced in #https://github.com/alan-turing-institute/sktime/pull/912, instead of the mixin we now use the tag `requires-fh-in-fit`

Reply:
is this `_tags` also inheritable so that I can write it only in `_Pipeline`? I think I can set here `_tags = {"requires-fh-in-fit": False}`  right?

Reply:
There is an ongoing discussion on tags in #981 but for now `{"requires-fh-in-fit": False}` seems to be correct

Reply:
well, I'd actually think it's `{requires-fh-in-fit": True}`, because formally (scitype level) it's whatever the forecaster at the end requires, and that could be `True` - in which case the pipeline would break. The "safe" option is hence always asking for `fh` in `fit`.

Reply:
>is this `_tags` also inheritable so that I can write it only in `_Pipeline`?

Yes, it's inheritable - use of the method `_all_tags` mimicks inheritance.

Reply:
But not all pipelines require fh in fit? Let's defer this discussion when we implement object-level tags (see #981)

Reply:
well, I think you need to decide now whether you want object level tags or not...
You can't be at the same time be for a tag with different value for a given concrete pipeline, and against dynamic tag override at construction. I.e., either we must say that `requires-fh-in-fit` is always true, or have a mechanism that inspects the tags of the pipeline at construction (or later) and assigns the right dynamic tag.

Reply:
new forecasting API was adapted by merge